### PR TITLE
Issue #2699747 Uses the FB Instant Articles SDK elements and validators in the fb_instant_articles_display module

### PR DESCRIPTION
**Pull request by m4olivei**
First crack at integrating the FB Instant Articles SDK elements to build up the Instant Article markup for a given node.

### CODE REVIEW

File: modules/fb_instant_articles_display/fb_instant_articles_display.module [hunk 1]
```diff
@@ -1,11 +1,34 @@
 <?php
-
```
Comment (scottrigby) [inline modules/fb_instant_articles_display/fb_instant_articles_display.module:2]: @m4olivei small style note, but let's add this extra linebreak back (I know Drupal core has some inconsistencies around this, but it's best practice to keep a space between the initial `<?php` and `@file` doxy block.

**Comment by m4olivei** [comment]
Rebased with @simonengelhardt's branch so that commit history is included here.

git rebase++

**Comment by scottrigby** [comment]
Glad to see the rebase @m4olivei ! I've been looking over both branches in the meantime, and will add a few quick comments now then more when I get back.

File: modules/fb_instant_articles_display/fb_instant_articles_display.module [hunk 2]
```diff
@@ -1,11 +1,50 @@
...
 /**
  * @file
  * Hook implementations for Facebook Instant Articles Display module.
  */
 
 /**
+* Implements hook_init().
+*/
+function fb_instant_articles_display_init() {
+ // Configure Instant Articles Transformer logger to append to Drupal log if enabled
```
Comment (scottrigby) [inline modules/fb_instant_articles_display/fb_instant_articles_display.module:15]: Let's add a TODO in the doc block here:

```
 * @todo Update to use a custom DrupalFBInstantArticlesLogger once
 *   @link https://github.com/facebook/facebook-instant-articles-sdk-php/issues/14 this SDK issue @endlink is in resolved.
```

File: README.txt [hunk 3]
```diff
@@ -21,9 +21,14 @@ This module requires the following modules:
 INSTALLATION
 ------------
 
+This module uses Composer to install dependencies. You can either execute
```
Comment (scottrigby) [inline README.txt]: Can we save the README update for https://github.com/BurdaMagazinOrg/module-fb_instant_articles/pull/19? I note more info about Composer options, and try to keep it simple there. Although I could add this link in part of the note that's already there.

File: .gitignore [hunk 4]
```diff
@@ -0,0 +1 @@
+/vendor/
```
Comment (scottrigby) [inline .gitignore]: Re the below comment, let's remove this gitignore file. Depending how you manage your dependencies, this makes some assumptions?

File: modules/fb_instant_articles_display/fb_instant_articles_display.module [hunk 5]
```diff
@@ -1,11 +1,50 @@
...
+ $enable_transformer_logging = variable_get('fb_instant_articles_display_enable_transformer_logging', FALSE);
```
Comment (vgardner) [inline modules/fb_instant_articles_display/fb_instant_articles_display.module:16]: We should delete this variable in the uninstall hook of the module.

File: modules/fb_instant_articles_display/fb_instant_articles_display.module [hunk 6]
```diff
@@ -6,6 +6,49 @@
  */
 
 /**
+* Implements hook_init().
+ *
+ * @todo Update to use a custom DrupalFBInstantArticlesLogger once
+ *   @link https://github.com/facebook/facebook-instant-articles-sdk-php/issues/14 this SDK issue @endlink is in resolved.
+*/
+function fb_instant_articles_display_init() {
```
Comment (vgardner) [inline modules/fb_instant_articles_display/fb_instant_articles_display.module:14]: Should this logger be configured on hook_init or only whenever the Transformer is being used?

File: modules/fb_instant_articles_display/fb_instant_articles_display.module [hunk 7]
```diff
@@ -6,6 +6,49 @@
...
+ // Configure Instant Articles Transformer logger to append to Drupal log if enabled
+ $enable_transformer_logging = variable_get('fb_instant_articles_display_enable_transformer_logging', FALSE);
```
Comment (vgardner) [inline modules/fb_instant_articles_display/fb_instant_articles_display.module:16]: We should delete this variable on the uninstall hook of the module.

File: modules/fb_instant_articles_display/fb_instant_articles_display.module (hunk 6 above)
Comment (m4olivei) [inline modules/fb_instant_articles_display/fb_instant_articles_display.module:14]: I don't know much about the Logger or what it's useful for.  I need to read up on this area to form an educated opinion about it's implementation.  I think @scottrigby has some opinions about this as well.

File: modules/fb_instant_articles_display/fb_instant_articles_display.module (hunk 7 above)
Comment (m4olivei) [inline modules/fb_instant_articles_display/fb_instant_articles_display.module:16]: Can do.  Added.

File: modules/fb_instant_articles_display/includes/InstantArticleNodeWrapper.inc [hunk 8]
```diff
@@ -0,0 +1,547 @@
+<?php
+
+/**
+ * @file
+ * Defines the InstantArticleNodeWrapper class.
+ */
+
```
Comment (scottrigby) [inline modules/fb_instant_articles_display/includes/InstantArticleNodeWrapper.inc:7]: Namespace :)

Comment (scottrigby) [inline modules/fb_instant_articles_display/includes/InstantArticleNodeWrapper.inc:7]: Actually let's save the class namespace for the follow-up PR you and I talked about (after this one is merged). So… never mind this comment for now :smile:

**Comment by scottrigby** [comment]
OK this is looking good :+1:  I'm going to go ahead and merge this. I spoke with @m4olivei and I plan to make a follow-up PR that moves some of this into an extension of the `\Facebook\InstantArticles\Elements\InstantArticle` class, so that similar hooks as the class in this PR invokes can be done by the base module (so that developers using the API module do not need to rely on the Display module for this). In that follow-up the class in this PR can then take advantage of that shared functionality. Coming soon :smiling_imp:

**Pull request merged**